web-common/src/features/dashboards/pivot/PivotTable.svelte

```diff
@@ -67,117 +98,93 @@
+  style:--row-height="{ROW_HEIGHT}px"
+  style:--header-length="{totalHeaderHeight}px"
   class="overflow-scroll h-fit max-h-full border rounded-md bg-white"
   bind:this={containerRefElement}
-  on:scroll={() => handleScroll(containerRefElement)}
 >
-  <table class="overflow-scroll">
-    <thead>
-      {#each headerGroups as headerGroup}
+  <div style:height="{totalRowSize + totalHeaderHeight}px">
+    <table>
+      <thead>
+        {#each headerGroups as headerGroup}
+          <tr>
+            {#each headerGroup.headers as header}
+              {@const sortDirection = header.column.getIsSorted()}
+              <th colSpan={header.colSpan}>
+                <div class="header-cell" style:height="{HEADER_HEIGHT}px">
+                  {#if !header.isPlaceholder}
+                    <button
+                      class="flex items-center gap-x-1"
+                      class:cursor-pointer={header.column.getCanSort()}
+                      class:select-none={header.column.getCanSort()}
+                      on:click={header.column.getToggleSortingHandler()}
+                    >
+                      {header.column.columnDef.header}
+                      {#if sortDirection}
+                        <span
+                          class="transition-transform -mr-1"
+                          class:-rotate-180={sortDirection === "desc"}
+                        >
+                          <ArrowDown />
+                        </span>
+                      {/if}
+                    </button>
+                  {:else}
+                    <button class="w-full h-full"></button>
+                  {/if}
+                </div>
+              </th>
+            {/each}
+          </tr>
+        {/each}
+      </thead>
+      <tbody>
         <tr>
-          {#each headerGroup.headers as header}
-            {@const sortDirection = header.column.getIsSorted()}
-            <th colSpan={header.colSpan}>
-              <div class="header-cell">
-                {#if !header.isPlaceholder}
-                  <button
-                    class="flex items-center gap-x-1"
-                    class:cursor-pointer={header.column.getCanSort()}
-                    class:select-none={header.column.getCanSort()}
-                    on:click={header.column.getToggleSortingHandler()}
-                  >
-                    {header.column.columnDef.header}
-                    {#if sortDirection}
-                      <span
-                        class="transition-transform -mr-1"
-                        class:-rotate-180={sortDirection === "desc"}
-                      >
-                        <ArrowDown />
-                      </span>
-                    {/if}
-                  </button>
-                {:else}
-                  <button class="w-full h-full"></button>
-                {/if}
-              </div>
-            </th>
-          {/each}
+          <td colspan={headerGroups.length} style:height="{before}px"> </td>
```

Comment:
Why do we need this? Wondering if we can follow Svelte virtual's example for sticky rows as well?
https://github.com/TanStack/virtual/blob/main/examples/svelte/sticky/src/App.svelte

Reply:
The rendering approach in this example uses absolute positioning of rows, which is not performant enough for our use case and does not work with well with rendering table elements as opposed to divs.

See: https://github.com/TanStack/virtual/discussions/476

See: https://github.com/TanStack/virtual/issues/585#issuecomment-1716247313

See: https://github.com/TanStack/virtual/issues/656